docs/plugins/haveibeenpwned/plugin.js

```diff
@@ -0,0 +1,250 @@
+                this.userEditView.$el.removeClass('input--error');
+            }
+        },
+        error: (e, xhr) => {
+            let err = xhr.response && xhr.response.error || new Error('Network error');
+            this.logger.error('Pwned Password API error', 'GET', xhr.status, err);
+            err.status = xhr.status;
+        }
+    });   
+};
+
+DetailsView.prototype.checkPwdPwned = function (passwordHash) {
+    this.logger.info('check hibp pwd (hash) ' + passwordHash);
+    
+    prefix = passwordHash.substring(0, 5);
+    new HIBPUtils().xhrcall({
+        url: `https://api.pwnedpasswords.com/range/${prefix}`,
+        method: 'GET',
+        responseType: 'text',
+        headers: undefined,
+        data: null,
+        statuses: [200, 404],
+        success: data => {
+            if (data) {
+                this.logger.info('found breaches ' + JSON.stringify(data));
+                data.split('\r\n').forEach(line => {
+                    h = line.split(':');
+                    suffix = h[0]; nb = h[1];
+                    if (prefix + suffix === passwordHash) {
+                        this._alert(`WARNING: This password is referenced as pawned ${nb} times on <a href="https://haveibeenpwned.com">https://haveibeenpwned.com</a>!\n`);
+                        this.passEditView.$el.focus();
+                        this.passEditView.$el.addClass('input--error');
+                        InputFx.shake(this.passEditView.$el);
+                    }
+                });
+            } else {
+                this.logger.info("check pwnd passwd passed...");
+                this.passEditView.$el.removeClass('input--error');
+            }    
+        },
+        error: (e, xhr) => {
+            let err = xhr.response && xhr.response.error || new Error('Network error');
+            this.logger.error('Pwned Password API error', 'GET', xhr.status, err);
+            err.status = xhr.status;
+        }
+    });
+};
+
+DetailsView.prototype.fieldChanged = function (e) {
+    //this.logger.info('field changed ' + new HIBPUtils().stringify(e));
+    detailsViewFieldChanged.apply(this, [e]);
+    if (e.field) {
+        if (e.field === '$Password' && this.checkPwnedPwd) {
+            if (this.passEditView.value) {
+                const pwd = this.passEditView.value.getText();
+                if (pwd.replace(/\s/, '') !== '' && !pwd.startsWith('{REF:')) {
+                    new HIBPUtils().sha1(pwd).then(hash => {
+                        this.checkPwdPwned(hash.toUpperCase());
+                    });
+                }
+            }
+        } else if (e.field === '$UserName' && this.checkPwnedName) {
+            this.checkNamePwned(e.val);
+        }
+    }
+};
+
+module.exports.getSettings = function () {
+    return [{
+        name: 'checkPwnedPwd',
+        label: 'Check passwords against HaveIBeenPwned list',
+        type: 'checkbox',
+        value: 'true'
```

Comment:
Please use boolean literal (`true`) instead of thuthy value (`'true'`) - the same below.

Reply:
done